src/Blocks/sources.jl

```diff
@@ -148,23 +254,64 @@ Generate exponentially damped sine signal.
 - `phase`: [rad] Phase of sine wave 
 - `offset`: Offset of output signal
 - `start_time`: [s] Output `y = offset` for `t < start_time`
+- `smooth`:  If `true`, returns a smooth wave. Defaults to `false`
 
 # Connectors:
 - `output`
 """
-function ExpSine(;name, 
+function ExpSine(; name, 
     frequency, 
     amplitude=1,
     damping=0.1,
     phase=0,
     offset=0,
-    start_time=0)
+    start_time=0,
+    smooth=false)
 
     @named output = RealOutput()
     pars = @parameters offset=offset start_time=start_time amplitude=amplitude frequency=frequency phase=phase damping=damping
+
+    equation = if smooth == false
+        offset + ifelse(t < start_time, 0, amplitude * exp(-damping * (t - start_time)) * sin(2*pi*frequency*(t - start_time) + phase))
+    else
+        δ = 1e-5
+        smooth_damped_sin(t, δ, frequency, amplitude, damping, phase, offset, start_time)
+    end
+    
+    eqs = [
+        output.u ~ equation
+    ]
+    compose(ODESystem(eqs, t, [], pars; name=name), [output])
+end
+
+"""
+Generate smooth triangular signal for frequencies less than or equal to 25 Hz
+
+# Parameters:
+- `frequency`: [Hz] Frequency of sqaure wave
+- `amplitude`: Amplitude of sqaure wave
+- `offset`: Offset of output signal
+- `start_time`: [s] Output `y = offset` for `t < start_time`
+
+# Connectors:
+- `output`
+"""
+function Triangular(; name, offset=0.0, amplitude=1.0,
+    frequency=1.0, start_time=0.0)
+    δ = 1e-5
+
+    @named output = RealOutput()
+    pars = @parameters begin
+        offset=offset
+        amplitude=amplitude 
+        frequency=frequency 
+        start_time=start_time 
+    end
+
     eqs = [
-        output.u ~ offset + ifelse(t < start_time, 0, amplitude * exp(-damping * (t - start_time)) * sin(2*pi*frequency*(t - start_time) + phase))
+        output.u ~ smooth_triangular(t, δ, frequency, amplitude, offset, start_time)
```

Comment:
Is there no exact version?

Reply:
When I had tried the solvers became unstable. Let me give another shot at it

Reply:
I would leave it in. It is an upstream problem than and not of the library. Also add tests but mark it as broken.